### PR TITLE
Add heartbeat as node sensor instance attribute

### PR DESCRIPTION
**Pull request by MartinHjelmare**
* Validate hearbeat payload as integer before casting to string.
* Update tests and handle internal message behavior.
* Add sensor attribute validation and more tests.

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
CC @inetAnt

**Comment by inetAnt** [comment]
Hey @MartinHjelmare,

I am testing this PR with a modified version of home-assistant (basically I just added the heartbeat state attribute for now).

When I watch home-assistant logs with a debug on mysensors and homeassistant.core.mysensors, I can see that my node is sending hearbeats but no state_changed events are fired : 

```
2018-05-28 23:32:23 DEBUG (MainThread) [mysensors] Receiving 3;255;3;0;22;2200
2018-05-28 23:32:35 DEBUG (MainThread) [mysensors] Receiving 3;255;3;0;22;2430
2018-05-28 23:32:46 DEBUG (MainThread) [mysensors] Receiving 3;255;3;0;22;2660
2018-05-28 23:32:57 DEBUG (MainThread) [mysensors] Receiving 3;255;3;0;22;2890
2018-05-28 23:33:08 DEBUG (MainThread) [mysensors] Receiving 3;255;3;0;22;3143
2018-05-28 23:33:17 DEBUG (MainThread) [mysensors] Receiving 1;0;1;0;1;65.5
2018-05-28 23:33:17 DEBUG (MainThread) [homeassistant.components.mysensors] Node update: node 1 child 0
```

I can see that heartbeat is always updated in the persistence file, but in home-assistant UI, the attribute only gets update when I have a value update from the node.

What should I see if `('_battery_level', '_heartbeat', '_protocol_version')` are changed?

**Comment by MartinHjelmare** [comment]
Right, I forgot we need to call `alert` method upon heartbeat to have pymysensors call the state update callback in home assistant. I'll add that asap.

File: tests/test_mysensors.py [hunk 1]
```diff
@@ -248,6 +248,35 @@ def test_bad_battery_level(gateway, add_sensor):
...
+
+def test_bad_heartbeat_value(gateway, add_sensor):
+    """Test internal receive of bad heartbeat."""
+    sensor = add_sensor(1)
+    gateway.logic('1;255;3;0;22;bad\n')
+    assert sensor.heartbeat == 0
+
+
+def test_set_bad_heartbeat(add_sensor):
+    """Test set a bad battery_level attribute on a node."""
```
Comment (MartinHjelmare) [inline tests/test_mysensors.py]: Stale docstring.

**Comment by MartinHjelmare** [comment]
@inetAnt should be fixed now.

**Comment by inetAnt** [comment]
@MartinHjelmare: awesome, Home Assistant is now seeing stuff, I am going to try patching it.

**Comment by MartinHjelmare** [comment]
Great! I'll merge this now.

**Pull request merged**